test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -840,6 +840,302 @@ public void TestTextLoaderBackCompat_VerWritt_0x0001000C()
+            VBuffer<Double> featuresPeriod = default;
+            ValueGetter<VBuffer<Double>> featuresDelegatePeriod = cursorPeriod.GetGetter<VBuffer<Double>>(columnsPeriod[1]);
+
+            UInt32[] labels = new uint[150];
+            double[][] features = new double[150][];
+
+            // Iterate over each row and save labels and features to array for future comparison
+            int count = 0;
+            while (cursorPeriod.MoveNext())
+            {
+                //Get values from respective columns
+                labelDelegatePeriod(ref labelPeriod);
+                featuresDelegatePeriod(ref featuresPeriod);
+                labels[count] = labelPeriod;
+                features[count] = featuresPeriod.GetValues().ToArray();
+                count++;
+            }
+
+            // Read dataset with comma as decimal marker.
+            string dataPathDecimalMarkerComma = GetDataPath("iris-decimal-marker-as-comma.txt");
+            var readerDecimalMarkerComma = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.UInt32, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
+                        },
+                DecimalMarker = ','
+            });
+            var textDataDecimalMarkerComma = readerDecimalMarkerComma.Load(GetDataPath(dataPathDecimalMarkerComma));
+
+            // Load values from iris-decimal-marker-as-comma.txt
+            DataViewSchema columnsComma = textDataDecimalMarkerComma.Schema;
+            using DataViewRowCursor cursorComma = textDataDecimalMarkerComma.GetRowCursor(columnsComma);
+            UInt32 labelComma = default;
+            ValueGetter<UInt32> labelDelegateComma = cursorComma.GetGetter<UInt32>(columnsComma[0]);
+            VBuffer<Double> featuresComma = default;
+            ValueGetter<VBuffer<Double>> featuresDelegateComma = cursorComma.GetGetter<VBuffer<Double>>(columnsComma[1]);
+
+            // Check values from iris-decimal-marker-as-comma.txt match those in iris.txt
+            count = 0;
+            while (cursorComma.MoveNext())
+            {
+                //Get values from respective columns
+                labelDelegateComma(ref labelComma);
+                featuresDelegateComma(ref featuresComma);
+                Assert.Equal(labels[count], labelComma);
+                Assert.Equal(features[count], featuresComma.GetValues().ToArray());
+                count++;
+            }
+        }
+
+        [Fact]
+        public void TestWrongDecimalMarkerInputs()
+        {
+            // When DecimalMarker does not match the actual decimal marker used in the dataset,
+            // we obtain values of NaN. Check that the values are indeed NaN in this case.
```

Comment:
Here you correctly checks for NaNs when using the wrong decimalMarker option for botch "iris.txt" (with .) and ""iris-decimal-marker-as-comma.txt" (with ,).

Is it possible to refactor this with InlineData() and a bool for (useComa) or something like that to avoid code repetition? I think it would be possible in this case since Columns are read as Single in both cases.

Reply:
Good idea!